File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length == 0){
            return 0;
        }
        Arrays.sort(nums);
        int currLen = 1;
        int max = 1;
        int lastEle = nums[0];
        for(int i=1; i<nums.length; i++){
            if(nums[i] == lastEle+1){
                currLen++;
                lastEle = nums[i];
                if(currLen > max){
                    max = currLen;
                }
            }
            else if(i > 0 && nums[i] == nums[i-1]){
                continue;
            }
            else{
                currLen = 1;
                lastEle = nums[i];
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]

        def rob_linear(houses):
            dp = [-1] * len(houses)

            def dfs(i):
                if i >= len(houses):
                    return 0
                if dp[i] != -1:
                    return dp[i]

                dp[i] = max(dfs(i+1), houses[i] + dfs(i+2))
                return dp[i]
        
            return dfs(0)
        
        return max(rob_linear(nums[:-1]), rob_linear(nums[1:]))
